src/keyboard.c

```diff
@@ -46,6 +46,7 @@ keyboard_modifiers_notify(struct wl_listener *listener, void *data)
 			desktop_focus_and_activate_view(&server->seat,
 				server->cycle_view);
 			desktop_move_to_front(server->cycle_view);
+			process_cursor_motion(server, event->time_msec);
```

Comment:
I've not had a chance to think this through, so am asking this as a question rather than a well thought-through statement.

1. Would it feel better to do the function all to update the cursor focus from desktop_move_to_front()? I think that's the function that moves windows around and therefore requires a cursor 'refocus'
2. If we call it from desktop_move_to_front(), we wouldn't have the event time, so would be better creating a wrapper function and use `clock_gettime(CLOCK_MONOTONIC, &now)` or similar. That's probably better anyway because it avoid using the time of the key event, which might be a bit before the windows had a chance to move around.
3. When making functions public, I like to use the filename as a prefix in the function name. So, might a good public function name for this wrapper be cursor_update_focus()?